### DISCUSSION TITLE
请教：如何获得浮层模块-右下角提示信息里面的内容或者错误类型

### DISCUSSION THREAD

**Discussion by hhh55555** [opensumi/core#3543] [Q&A] [answered]
<img width="861" alt="image" src="https://github.com/opensumi/core/assets/64674757/a7f276f7-be99-48ab-bd9a-251bd250f74e">
使用的是纯前端的插件，初次点击目录树的每个文件时，都会调用我们这边的接口，如果之前打开过的话，就不调用了。
最近自测时发现偶尔会报错，经过复现：打开文件报错，主要有以下几种情况：
1、接口正常返回，但是确提示local storage为null
2、接口显示302被重定向，导致没办法返回数据，报错显示文件length为0 ，上面两种情况均能通过清空缓存刷新后解决。我想在右下角报错的时候，清空本地的local storage，然后强制刷新解决上面这两种情况，但是现在不知道怎么获得右下角的内容或者提示，我看源码是有editor.failToOpen这种错误类型的，要是我这边能感知到就好了

**Comment by erha19** [comment, marked as answer]
@hhh55555 这块代码相关的逻辑在这里，你可以增加一个 Event ，后续可通过 Event 来进行后续处理，暂时框架内少处理了这种情况。

https://github.com/opensumi/core/blob/0c8a216fa2be1292d69faf05e7d4a15dd4cfcaa3/packages/editor/src/browser/workbench-editor.service.ts#L1482-L1486
